sklearn/metrics/_regression.py

```diff
@@ -189,6 +190,77 @@ def mean_absolute_error(y_true, y_pred,
     return np.average(output_errors, weights=multioutput)
 
 
+def mean_absolute_percentage_error(y_true, y_pred,
+                                   sample_weight=None,
+                                   multioutput='uniform_average'):
+    """Mean absolute percentage error regression loss
+
+    Read more in the :ref:`User Guide <mean_absolute_percentage_error>`.
+
+    Parameters
+    ----------
+    y_true : array-like of shape = (n_samples,) or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+
+    y_pred : array-like of shape = (n_samples,) or (n_samples, n_outputs)
+        Estimated target values.
+
+    sample_weight : array-like of shape = (n_samples,), optional
+        Sample weights.
+
+    multioutput : string in ['raw_values', 'uniform_average'] or list.
+        Defines aggregating of multiple output values.
+        Array-like value defines weights used to average errors.
+        If input is list then the shape must be (n_outputs,).
+
+        'raw_values' :
+            Returns a full set of errors in case of multioutput input.
+
+        'uniform_average' :
+            Errors of all outputs are averaged with uniform weight.
+
+
+    Returns
+    -------
+    loss : float or ndarray of floats
+        If multioutput is 'raw_values', then mean absolute percentage error
+        is returned for each output separately.
+        If multioutput is 'uniform_average' or an ndarray of weights, then the
+        weighted average of all output errors is returned.
+
+        MAPE output is non-negative floating point. The best value is 0.0.
+
+    Examples
+    --------
+    >>> from sklearn.metrics import mean_absolute_percentage_error
+    >>> y_true = [3, -0.5, 2, 7]
+    >>> y_pred = [2.5, 0.0, 2, 8]
+    >>> mean_absolute_percentage_error(y_true, y_pred)
+    32.73...
+    >>> y_true = [[0.5, 1], [-1, 1], [7, -6]]
+    >>> y_pred = [[0, 2], [-1, 2], [8, -5]]
+    >>> mean_absolute_percentage_error(y_true, y_pred)
+    55.15...
+    >>> mean_absolute_percentage_error(y_true, y_pred, multioutput=[0.3, 0.7])
+    61.98...
+    """
+    y_type, y_true, y_pred, multioutput = _check_reg_targets(
+        y_true, y_pred, multioutput)
+    check_consistent_length(y_true, y_pred, sample_weight)
+    epsilon = np.finfo(float).eps
```

Comment:
Please put the precision explicitly:  `np.finfo(np.float64).eps`.

Actually I wonder if we should make it depend on the dtype of `y_true`:

```python
epsilon = np.finfo(y_true.dtype).eps
```

Any opinion? Maybe always using `np.float64` is less likely to cause hard to understand, unexpected behaviors though.

Reply:
I agree with your opinion. It's good to have dtype of y_true than taking float64 for every case.

Reply:
I think np.float64 makes more sense to me because in most of other datatype cases, the datatype would be integer, and it will fail finding the eps for that since eps has to be float real number. @ogrisel, Your comments will be highly appreciated on this.

Reply:
Let keep `np.finfo(np.float64).eps` then.

Reply:
@ogrisel  ,I have kept np.float64 in current commit. Please review the complete code then.